test/basic_parser.cpp

```diff
@@ -690,14 +692,22 @@ class basic_parser_test
     {
         std::vector<parse_options> all_configs =
         {
-            make_options(false, false, true),
-            make_options(true, false, true),
-            make_options(false, true, true),
-            make_options(true, true, true),
-            make_options(false, false, false),
-            make_options(true, false, false),
-            make_options(false, true, false),
-            make_options(true, true, false)
+            make_options(false, false, true, false),
+            make_options(true, false, true, false),
+            make_options(false, true, true, false),
+            make_options(true, true, true, false),
+            make_options(false, false, false, false),
+            make_options(true, false, false, false),
+            make_options(false, true, false, false),
+            make_options(true, true, false, false),
+            make_options(false, false, true, true),
+            make_options(true, false, true, true),
+            make_options(false, true, true, true),
+            make_options(true, true, true, true),
+            make_options(false, false, false, true),
+            make_options(true, false, false, true),
+            make_options(false, true, false, true),
+            make_options(true, true, false, true)
```

Comment:
These would look better using 0 and 1 instead of true and false I bet.. lol

Reply:
Given the growing amount of combinations, I'd rather replace it with a loop (see https://github.com/boostorg/json/pull/657/commits/d665b08517a9d9bbc4475003eb4b5cfa2d920579).

Reply:
The loop is smaller but it is easier to understand and read when it is written out. And it is easier to set a breakpoint on a particular combination without the loop..

Reply:
Ok, done